File: analyze_excel.py
#!/usr/bin/env python3
"""
Analyze datamigration.xlsx - list sheets, headers, sample data
"""
import openpyxl
from openpyxl.utils import get_column_letter

FILE = '/Users/a1/Documents/motivation/fundamental-engine/datamigration.xlsx'
MAX_SAMPLE_ROWS = 5
MAX_COLS = 30

def cell_val(cell):
    if cell is None:
        return ''
    v = cell.value
    if v is None:
        return ''
    return str(v)[:80]

wb = openpyxl.load_workbook(FILE, read_only=True, data_only=True)
sheets = wb.sheetnames
print(f'=== TOTAL SHEETS: {len(sheets)} ===\n')
for i, name in enumerate(sheets):
    print(f'{i+1:2}. [{name}]')

print('\n' + '='*80)

for sheet_name in sheets:
    ws = wb[sheet_name]
    print(f'\n{"="*80}')
    print(f'SHEET: [{sheet_name}]')
    print(f'{"="*80}')
    
    rows_read = 0
    for row in ws.iter_rows(max_row=MAX_SAMPLE_ROWS + 3, max_col=MAX_COLS):
        if rows_read >= MAX_SAMPLE_ROWS:
            break
        row_vals = [cell_val(c) for c in row]
        # skip entirely empty rows
        if not any(v.strip() for v in row_vals):
            continue
        # trim trailing empty
        while row_vals and not row_vals[-1].strip():
            row_vals.pop()
        print(' | '.join(f'{v:<20}' for v in row_vals))
        rows_read += 1
    
    print(f'  (max_row≈{ws.max_row}, max_col≈{ws.max_column})')

wb.close()
print('\n=== DONE ===')

File: analyze_excel.js
#!/usr/bin/env node
// analyze_excel.js — Node.js script to analyze datamigration.xlsx
// Usage: node analyze_excel.js (requires xlsx module installed)
const path = require('path');
const fs = require('fs');

let XLSX;
try {
  XLSX = require('./node_modules/xlsx');
} catch (e) {
  try {
    XLSX = require('xlsx');
  } catch (e2) {
    console.error('xlsx module not found. Run: npm install xlsx');
    process.exit(1);
  }
}

const FILE = path.join(__dirname, 'datamigration.xlsx');
const wb = XLSX.readFile(FILE, { cellFormula: false, sheetRows: 8 });

console.log(`=== TOTAL SHEETS: ${wb.SheetNames.length} ===\n`);
wb.SheetNames.forEach((name, i) => {
  console.log(`${String(i+1).padStart(2)}. [${name}]`);
});

console.log('\n' + '='.repeat(80));

wb.SheetNames.forEach(name => {
  const ws = wb.Sheets[name];
  const range = XLSX.utils.decode_range(ws['!ref'] || 'A1:A1');
  const data = XLSX.utils.sheet_to_json(ws, { header: 1, defval: '', blankrows: false });
  
  console.log(`\n${'='.repeat(80)}`);
  console.log(`SHEET: [${name}]   rows≈${range.e.r+1}  cols≈${range.e.c+1}`);
  console.log('='.repeat(80));
  
  const MAX_ROWS = 6;
  let count = 0;
  for (const row of data) {
    if (count >= MAX_ROWS) break;
    const trimmed = row.slice(0, 25).map(v => String(v).slice(0, 25));
    if (trimmed.every(v => v.trim() === '')) continue;
    // trim trailing empty
    let last = trimmed.length - 1;
    while (last >= 0 && trimmed[last].trim() === '') last--;
    console.log(trimmed.slice(0, last+1).map(v => v.padEnd(25)).join(' | '));
    count++;
  }
});

console.log('\n=== DONE ===');
